src/controller/expenses.controller.js

```diff
@@ -0,0 +1,107 @@
+const expensesService = require('../services/expenses.service');
+const userService = require('../services/user.service');
+
+const get = (req, res) => {
+  const { userId, from, to, categories } = req.query;
+
+  if (!userId && !categories && (!from || !to)) {
```

Comment:
This condition will only return all expenses if `userId`, `categories`, `from` and `to` are all not provided. If only some of them are provided, the function will continue and may filter the expenses based on the provided parameters. This may not be what you intended.